### PR TITLE
Merge CPU/GPU nuget pipeline

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

Merge CPU/GPU nuget pipeline.  The old GPU nuget pipeline will be only for DML. 

TODO: the result GPU package contains PDB files for some of the DLLs, but not all. It is due to the refactoring of CUDA EP to pluggable DLLs. At that time we forgot to copy the PDB files.  However, I can't add them in now.  Because currently the package is already 220MB large. If the missed PDB files were added, then it will be oversize. nuget.org doesn't accept >250MB packages. 

Therefore I suggest we **delete** all the PDB files and find another way to handle it.

**Motivation and Context**
- Why is this change required? What problem does it solve?

Finish compliance work items. The old GPU one misses compliance steps.  And it forgot to copy \*.pdb files for onnxruntime\_providers\_\*.dll to the packages.  Enable the GPU tests.

- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.bat [hunk 1]
```diff
@@ -39,14 +39,19 @@ IF EXIST test\Microsoft.ML.OnnxRuntime.EndToEndTests\obj RMDIR /S /Q test\Micros
...
 
 IF NOT errorlevel 0 (
     @echo "Failed to restore nuget packages for the test project"
     EXIT 1
 )
 
-%dn% test test\Microsoft.ML.OnnxRuntime.EndToEndTests\Microsoft.ML.OnnxRuntime.EndToEndTests.csproj --no-restore
+IF "%PACKAGENAME%"=="Microsoft.ML.OnnxRuntime.Gpu" (
+  set TESTONGPU=ON
+  %dn% test -p:DefineConstants=USE_CUDA test\Microsoft.ML.OnnxRuntime.EndToEndTests\Microsoft.ML.OnnxRuntime.EndToEndTests.csproj --no-restore
```
Comment (snnn) [inline csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.bat:51]: If TRT is also in this package, we will add TRT here too.

Comment (chilo-ms) [reply csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.bat:51]: Yes, we will need to add TRT defineConstants here too. I'll take care of this after this PR is merged first.

**Comment by snnn** [comment]
The "Nuget WindowsAI Pipeline" build failure is not related to my change. WindowsAI team is fixing it.

File: tools/ci_build/github/windows/bundle_dlls_gpu.bat [hunk 2]
```diff
@@ -0,0 +1,12 @@
+REM for available runtime identifiers, see https://github.com/dotnet/corefx/blob/release/3.1/pkg/Microsoft.NETCore.Platforms/runtime.json
```
Comment (pranavsharma) [inline tools/ci_build/github/windows/bundle_dlls_gpu.bat:1]: Add license header.

File: tools/ci_build/github/windows/extract_nuget_files_gpu.ps1 [hunk 3]
```diff
@@ -0,0 +1,31 @@
+New-Item -Path $Env:BUILD_BINARIESDIRECTORY\RelWithDebInfo\RelWithDebInfo\nuget-artifacts -ItemType directory
```
Comment (pranavsharma) [inline tools/ci_build/github/windows/extract_nuget_files_gpu.ps1:1]: Add license header.

**Review by pranavsharma — APPROVED** [review]

**Pull request merged**